docs/machine-learning/how-does-mldotnet-work.md

```diff
@@ -0,0 +1,276 @@
+
+### Data models and schema
+
+At the core of an ML.NET machine learning pipeline are [DataView](xref:Microsoft.ML.IDataView) objects.
+
+Each transformation in the pipeline has an input schema (data names, types, and sizes that the transform expects to see on its input); and an output schema (data names, types, and sizes that the transform produces after the transformation). 
+
+If the output schema from one transform in the pipeline does not match the input schema of the next transform, ML.NET will throw an exception.
+
+A data view object has columns and rows. Each column has a name and a type and a length. For example: the input columns in the house price example are **Size** and **Price**. They are both type  and they are scalar quantities rather than vector ones.
+
+   ![ML.NET Data View example with house price prediction data](./media/ml-net-dataview.png)
+
+All ML.NET algorithms look for an input column that is a vector. By default this vector column is called **Features**. This is why we concatenated the **Size** column into a new column called **Features** in our house price example.
+
+ ```csharp
+    var pipeline = mlContext.Transforms.Concatenate("Features", new[] { "Size" })
+ ```
+
+All algorithms also create new columns after they have performed a prediction. The fixed names of these new columns depend on the type of machine learning algorithm. For the regression task, one of the new columns is called **Score**. This is why we attributed our price data with this name.
+
+```csharp
+    public class Prediction
+    {
+        [ColumnName("Score")]
+        public float Price { get; set; }
+    }
+```    
+
+You can find out more about output columns of different machine learning tasks in the [Machine Learning Tasks](resources/tasks.md) guide.
+
+An important property of DataView objects is that they are evaluated **lazily**. Data views are only loaded and operated on during model training and evaluation, and data prediction. While you are writing and testing your ML.NET application, you can use the Visual Studio debugger to take a peek at any data view object by calling the [Preview](xref:Microsoft.ML.DebuggerExtensions.Preview*) method.
+
+```csharp
+    var debug = testPriceDataView.Preview();
+```
+
+You can watch the `debug` variable in the debugger and examine its contents. Do not use the Preview method in production code, as it significantly degrades performance.
+
+### Model Deployment
+
+In real-life applications, your model training and evaluation code will be separate from your prediction. In fact, these two activities are often performed by separate teams. Your model development team can save the model for use in the prediction application.
+
+```csharp   
```

Comment:
Suggestion: This snippet can be simplified since the model is being saved to a local file. Use the method that takes in the string path as opposed to a Stream.

```csharp
mlContext.Model.Save(model, trainingData.Schema,"model.zip");
```

Reply:
Well-spotted. Fixing, thanks!